ClientLib/src/main/java/com/o3dr/android/client/utils/data/tlog/TLogParser.java

```diff
@@ -0,0 +1,349 @@
+package com.o3dr.android.client.utils.data.tlog;
+
+import android.os.Handler;
+import android.util.Log;
+
+import com.MAVLink.MAVLinkPacket;
+import com.MAVLink.Messages.MAVLinkMessage;
+import com.MAVLink.Parser;
+
+import java.io.BufferedInputStream;
+import java.io.DataInputStream;
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URI;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Parse TLog file into Events
+ */
+public class TLogParser {
+    private static final String LOG_TAG = TLogParser.class.getSimpleName();
+
+    private static final Parser parser = new Parser();
+    private static ExecutorService parseTlogExecutor;
+
+    public static class TLogIterator {
+        private File file;
+        private DataInputStream in = null;
```

Comment:
@chaviw no need to declare it with a `null` value here. It's done automatically.